core/templates/pages/admin-page/activities-tab/admin-dev-mode-activities-tab.component.spec.ts

```diff
@@ -0,0 +1,415 @@
+
+describe('Admin dev mode activities tab', () => {
+  let component: AdminDevModeActivitiesTabComponent;
+  let fixture: ComponentFixture<AdminDevModeActivitiesTabComponent>;
+  let adminBackendApiService: AdminBackendApiService;
+  let adminDataService: AdminDataService;
+  let adminTaskManagerService: AdminTaskManagerService;
+  let windowRef: WindowRef;
+  let adminDataObject = {
+    demoExplorationIds: ['expId'],
+    demoExplorations: [
+      [
+        '0',
+        'welcome.yaml'
+      ]
+    ],
+    demoCollections: [
+      ['collectionId']
+    ],
+  } as AdminPageData;
+  let mockConfirmResult: (val: boolean) => void;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        MaterialModule,
+        FormsModule,
+        HttpClientTestingModule
+      ],
+      declarations: [
+        AdminDevModeActivitiesTabComponent
+      ]
+    }).compileComponents();
+  }));
+
+  beforeEach(async(() => {
+    fixture = TestBed.createComponent(AdminDevModeActivitiesTabComponent);
+    component = fixture.componentInstance;
+    adminBackendApiService = TestBed.get(AdminBackendApiService);
+    adminDataService = TestBed.get(AdminDataService);
+    adminTaskManagerService = TestBed.get(AdminTaskManagerService);
+    windowRef = TestBed.get(WindowRef);
+
+    spyOn(adminDataService, 'getDataAsync').and.resolveTo(adminDataObject);
+
+    let confirmResult = true;
+    spyOnProperty(windowRef, 'nativeWindow').and.returnValue({
+      confirm: () => confirmResult
+    });
+
+    mockConfirmResult = (val) => {
+      confirmResult = val;
+    };
+
+    fixture.detectChanges();
+    component.ngOnInit();
+  }));
+
+  it('should create', () => {
+    expect(component).toBeDefined();
+  });
+
+  it('should initialize data correctly', () => {
+    expect(component.reloadingAllExplorationPossible).toEqual(true);
+    expect(component.demoExplorationIds).toEqual(['expId']);
+    expect(component.DEMO_COLLECTIONS).toEqual([['collectionId']]);
+    expect(component.DEMO_EXPLORATIONS).toEqual([['0', 'welcome.yaml']]);
+  });
+
+  describe('.reloadExploration', () => {
+    it('should not reload a specific exploration if task is running', () => {
+      spyOn(
+        adminTaskManagerService, 'isTaskRunning').and.returnValue(true);
+      expect(component.reloadExploration('expId')).toBeUndefined();
+    });
+
+    it('should not procees if user doesn\'t confirm', () => {
+      mockConfirmResult(false);
+      expect(component.reloadExploration('expId')).toBeUndefined();
+    });
+
+    it('should load explorations', async(() => {
+      const expId = component.demoExplorationIds[0];
+
+      spyOn(adminBackendApiService, 'reloadExplorationAsync')
+        .and.returnValue(Promise.resolve());
+      spyOn(adminTaskManagerService, 'isTaskRunning').and.returnValue(false);
+      spyOn(component.setStatusMessage, 'emit');
+
+      component.reloadExploration(expId);
```

Comment:
Do we also need to assert the `Proccessing....` message?

Reply:
We can. Added the tests for the same.